plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -445,6 +447,67 @@ func EqualPriorityMap(_ *v1.Pod, _ interface{}, nodeInfo *schedulercache.NodeInf
 	}, nil
 }
 
+// pickOneNodeForPreemption chooses one node among the given nodes. It assumes
+// pods in each map entry are ordered by their priority.
+// It picks a node based on the following criteria:
+// 1. A node with minimum highest priority victim is picked.
+// 2. Ties are broken by sum of priorities of all victims.
+// 3. If there are still ties, node with the minimum number of victims is picked.
+// 4. If there are still ties, the first such node in the map is picked (sort of randomly).
+func pickOneNodeForPreemption(nodesToPods map[string][]*v1.Pod) string {
+	type nodeScore struct {
+		highestPriority int32
+		sumPriorities   int64
+		numPods         int
```

Comment:
math.MaxInt32 is, unlike its name, of type int. We also assign len() to numPods which is also of type int. That's why numPods is an int.